src/modes/draw_polygon.js

```diff
@@ -0,0 +1,91 @@
+var {isEnterKey, isEscapeKey} = require('../lib/common_selectors');
+var Polygon = require('../feature_types/polygon');
+
+module.exports = function(ctx) {
+
+  var geojson = {
+    "type": "Feature",
+    "properties": {},
+    "geometry": {
+      "type": "Polygon",
+      "coordinates": [[[0,0], [0, 0],[0, 0]]]
+    }
+  }
+
+  var feature = new Polygon(ctx, geojson);
+
+  var stopDrawingAndRemove = function() {
+    ctx.events.changeMode('default');
+    ctx.store.delete(feature.id);
+  }
+
+  var pos = 0;
+
+  var onMouseMove = function(e) {
+    if(pos === 0) {
+      feature.updateCoordinate(`0.${0}`, e.lngLat.lng, e.lngLat.lat);
+      feature.updateCoordinate(`0.${1}`, e.lngLat.lng, e.lngLat.lat);
```

Comment:
The point here is to match what `queryRenderFeatures` returns. With VT the properties are either a number or a string. I assumed that was the same for GeoJSON sent to GL, but I didn't check this. I wonder if we could send an array.